Calibration/HcalIsolatedTrackReco/src/IsolatedPixelTrackCandidateL1TProducer.cc

```diff
@@ -0,0 +1,315 @@
+  tauAssocCone_(                config.getParameter<double>("tauAssociationCone") ), 
+  tauUnbiasCone_(               config.getParameter<double>("tauUnbiasCone") ),
+  minPTrackValue_(              config.getParameter<double>("minPTrack") ),
+  maxPForIsolationValue_(       config.getParameter<double>("maxPTrackForIsolation") ),
+  ebEtaBoundary_(               config.getParameter<double>("EBEtaBoundary") ),
+  rEB_( -1 ), 
+  zEE_( -1 ),
+  bfVal_( 0 )
+{
+  // Register the product
+  produces< reco::IsolatedPixelTrackCandidateCollection >();
+}
+
+IsolatedPixelTrackCandidateL1TProducer::~IsolatedPixelTrackCandidateL1TProducer() {
+
+}
+
+void IsolatedPixelTrackCandidateL1TProducer::beginRun(const edm::Run &run, const edm::EventSetup &theEventSetup) {
+
+  edm::ESHandle<CaloGeometry> pG;
+  theEventSetup.get<CaloGeometryRecord>().get(pG);   
+  
+  const double rad (dynamic_cast<const EcalBarrelGeometry*>( pG->getSubdetectorGeometry(DetId::Ecal, EcalBarrel ))->avgRadiusXYFrontFaceCenter() ) ;
+  const double zz  (dynamic_cast<const EcalEndcapGeometry*>( pG->getSubdetectorGeometry(DetId::Ecal, EcalEndcap ))->avgAbsZFrontFaceCenter() ) ;
+
+  rEB_ = rad;
+  zEE_ = zz;
+
+  edm::ESHandle<MagneticField> vbfField;
+  theEventSetup.get<IdealMagneticFieldRecord>().get(vbfField);
+  const VolumeBasedMagneticField* vbfCPtr = dynamic_cast<const VolumeBasedMagneticField*>(&(*vbfField));
+  GlobalVector BField = vbfCPtr->inTesla(GlobalPoint(0,0,0));
+  bfVal_=BField.mag();
+}
+
+void IsolatedPixelTrackCandidateL1TProducer::produce(edm::Event& theEvent, const edm::EventSetup& theEventSetup) {
+
+  reco::IsolatedPixelTrackCandidateCollection* trackCollection=new reco::IsolatedPixelTrackCandidateCollection;
+
+  //create vector of refs from input collections
+  std::vector<reco::TrackRef> pixelTrackRefs;
+
+  for (unsigned int iPix=0; iPix<toks_pix_.size(); iPix++) {
+    edm::Handle<reco::TrackCollection> iPixCol;
+    theEvent.getByToken(toks_pix_[iPix],iPixCol);
+    for (reco::TrackCollection::const_iterator pit=iPixCol->begin(); pit!=iPixCol->end(); pit++) {
+      pixelTrackRefs.push_back(reco::TrackRef(iPixCol,pit-iPixCol->begin()));
+    }
+  }
+
+  edm::Handle<l1t::JetBxCollection> l1eTauJets;
+  theEvent.getByToken(tok_l1_,l1eTauJets);
+
+  edm::Handle<reco::VertexCollection> pVert;
+  theEvent.getByToken(tok_vert_,pVert);
+
+  double ptTriggered  = -10;
+  double etaTriggered = -100;
+  double phiTriggered = -100;
+  
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> l1trigobj;
+  theEvent.getByToken(tok_hlt_, l1trigobj);
+  
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1tauobjref;
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1jetobjref;
+  std::vector< edm::Ref<l1t::JetBxCollection> > l1forjetobjref;
+  
+  l1trigobj->getObjects(trigger::TriggerL1TauJet, l1tauobjref);
+  l1trigobj->getObjects(trigger::TriggerL1CenJet, l1jetobjref);
+  l1trigobj->getObjects(trigger::TriggerL1ForJet, l1forjetobjref);
+  
```

Comment:
Cen and For jet no longer exists.
The enums have become TriggerTau and TriggerJet